src/lib.rs

```diff
@@ -1341,9 +1341,11 @@ pub fn z85_decode(data: &str) -> result::Result<Vec<u8>, DecodeError> {
     let mut dest = vec![0u8; len];
 
     let c_str = ffi::CString::new(data)?;
+    let c_str_ptr = c_str.into_raw();
 
     unsafe {
-        zmq_sys::zmq_z85_decode(dest.as_mut_ptr(), c_str.into_raw());
+        zmq_sys::zmq_z85_decode(dest.as_mut_ptr(), c_str_ptr);
+        drop(ffi::CString::from_raw(c_str_ptr));
```

Comment:
Any reason for not simply using https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr, and remove `into_raw`/`from_raw` altogether?

Besides, the `drop()` here is superfluous.

Reply:
Good point—I'll update to use `as_ptr`.